homeassistant/components/smappee_official/sensor.py

```diff
@@ -0,0 +1,363 @@
+
+                entities.append(
+                    SmappeeSensor(
+                        smappee_base=smappee_base,
+                        service_location=service_location,
+                        sensor="sensor",
+                        attributes=[
+                            channel.get("name"),
+                            gw_icon,
+                            channel.get("uom"),
+                            f"{sensor_id}-{channel.get('channel')}",
+                            None,
+                        ],
+                    )
+                )
+
+    async_add_entities(entities, True)
+
+
+class SmappeeSensor(Entity):
+    """Implementation of a Smappee sensor."""
+
+    def __init__(self, smappee_base, service_location, sensor, attributes):
+        """Initialize the Smappee sensor."""
+        self._smappee_base = smappee_base
+        self._service_location = service_location
+        self._sensor = sensor
+        self.data = None
+        self._state = None
+        self._name = attributes[0]
+        self._icon = attributes[1]
+        self._unit_of_measurement = attributes[2]
+        self._sensor_id = attributes[3]
+        self._device_class = attributes[4]
+
+    @property
+    def name(self):
+        """Return the name for this sensor."""
+        if self._sensor in ["sensor", "load"]:
+            return (
+                f"{self._service_location.service_location_name} - "
+                f"{self._sensor.title()} - {self._name}"
+            )
+
+        return f"{self._service_location.service_location_name} - {self._name}"
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_class(self):
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return self._device_class
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def unique_id(self,):
+        """Return the unique ID for this sensor."""
+        if self._sensor in ["load", "sensor"]:
+            return (
+                f"{self._service_location.device_serial_number}-"
+                f"{self._service_location.service_location_id}-"
+                f"{self._sensor}-{self._sensor_id}"
+            )
+
+        return (
+            f"{self._service_location.device_serial_number}-"
+            f"{self._service_location.service_location_id}-"
+            f"{self._sensor}"
+        )
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        attributes = {
+            "Service location name": self._service_location.service_location_name,
+            "Device serialnumber": self._service_location.device_serial_number,
```

Comment:
Do we need the serial number in the state attributes?

Reply:
Removed all state attributes